app/gui2/src/components/GraphEditor/widgets/WidgetNumber.vue

```diff
@@ -10,9 +10,9 @@ import { computed, ref, type ComponentInstance } from 'vue'
 const props = defineProps(widgetProps(widgetDefinition))
 const inputComponent = ref<ComponentInstance<typeof NumericInputWidget>>()
 
-function setValue(value: number | string) {
+function setValue(value: string | undefined) {
   props.onUpdate({
-    portUpdate: { value: value.toString(), origin: props.input.portId },
+    portUpdate: { value: value ?? '', origin: props.input.portId },
```

Comment:
`undefined` probably should be propagated as such, as it correctly signals a removal of an argument. Using empty string here can introduce issues by retaining unnecessary extra spaces.